### DISCUSSION TITLE
Cleanup after terminating workflow

### DISCUSSION THREAD

**Discussion by Sergey-Terekhin** [danielgerlag/workflow-core#962] [Q&A]
Hi all!
Thanks for the great library!

My question is about termination workflow: does any (preferably simple) way exist to compensate executed steps in terminated workflow? 

For example, I create workflow, which consists from 5 steps and every step creates some file. When I call Terminate, I'd like to delete created files somehow. For now I have the only idea: to create clean-up workflow and start it after termination the original one. 

For me specifying cleanup action for every step and execute them in reverse order would be more preferable, but for now it's impossible as I understand

Any other solution?

**Comment by christopher-watanabe-snkeos** [comment]
Hi Sergey, 

I'm a relative noob when it comes to workflow-core, but it sounds like what you might be looking for is post-workflow Middleware, for example the printing of workflow step time information given in the project's [readthedocs](https://workflow-core.readthedocs.io/en/latest/workflow-middleware/). Take a look at the _PrintWorkflowSummaryMiddleware_, it might guide you to buliding a custom method that deletes files based on the step? Or if you store the files in a known location, you can simply remove all files in this location post-workflow.
